mne/encoding/model.py

```diff
@@ -0,0 +1,238 @@
+        # Return masked values
+        if y is not None:
+            return X, y
+        return X
+
+    def transform(self, X, y=None):
+        """Return a subset of data points.
+
+        Returns
+        -------
+        data : array, shape (n_masked_samples, n_features)
+            A subset of rows from the input array.
+        """
+        return self._mask_data(X, y)
+
+    @property
+    def _final_estimator(self):
+        return _get_final_est(self)
+
+
+def _get_final_est(est):
+    """Return the final component of a sklearn estimator/pipeline.
+
+    Parameters
+    ----------
+    est : pipeline, or sklearn / MNE estimator
+        An estimator chain from which you wish to pull the final estimator.
+
+    Returns
+    -------
+    est : the sklearn-style estimator at the end of the input chain.
+    """
+    # Define classes where we pull `est` manually
+    import sklearn
+    from distutils.version import LooseVersion
+    if LooseVersion(sklearn.__version__) < '0.16':
+        raise ValueError('Encoding models require sklearn version >= 0.16')
+
+    iter_classes = (SampleMasker,)
+    # Iterate in case we have meta estimators w/in meta estimators
+    while hasattr(est, '_final_estimator') or isinstance(est, iter_classes):
+        if isinstance(est, iter_classes):
+            est = est.est
+        else:
+            est = est._final_estimator
+    return est
+
+
+def get_coefs(est, coef_name='coef_'):
+    """Pull coefficients from an estimator object.
+
+    Parameters
+    ----------
+    est : a sklearn estimator
+        The estimator from which to pull the coefficients.
+    coef_name : string
+        The name of the attribute corresponding to the coefficients created
+        after fitting. Defaults to 'coef_'
+
+    Returns
+    -------
+    coefs : array, shape (n_targets, n_coefs)
+        The output coefficients.
+    """
+    est = _get_final_est(est)
+    if not hasattr(est, coef_name):
+        raise ValueError('Estimator either is not fit or does not use'
+                         ' coefficient name: %s' % coef_name)
+    coefs = getattr(est, coef_name)
+    return coefs
+
+
+def _check_estimator(est):
+    from ..decoding.base import _check_estimator as _dec_check_estimator
+    """Ensure the estimator will work for regression data."""
+    from sklearn.linear_model import Ridge
+    from sklearn.pipeline import Pipeline
+
+    # Define string-based solvers
+    _ridge_solvers = ['auto', 'svd', 'cholesky', 'lsqr', 'sparse_cg', 'sag']
+
+    est = 'auto' if est is None else est
+    if est in _ridge_solvers:
+        est = Ridge(alpha=0, solver=est, fit_intercept=False)
+    elif isinstance(est, string_types):
+        raise ValueError("estimator must be a scikit-learn estimator or one of"
+                         " {}".format(_ridge_solvers))
+    else:
+        _dec_check_estimator(est)
+
+    # Make sure we have a pipeline
+    if not isinstance(est, Pipeline):
```

Comment:
Was a carryover from the old rERP days way back when I was young and inexperienced...removing.